src/lib/core/components/computations/ComputeJobStatusHook.ts

```diff
@@ -45,8 +66,6 @@ export function useComputeJobStatus(
   useEffect(() => {
     // Track if effect has been "cancelled"
     let cancelled = false;
-    // Clear existing job status
-    setJobStatus(undefined);
```

Comment:
So the key part is not doing this inside the useEffect, but instead above at line 54?

Reply:
Yes! This is the key change for fixing the bug. The rest is either cleanup or code to support detecting when we need to set the job status to `undefined`.